File: solutions/ruby/secret-handshake/1/secret_handshake.rb
class SecretHandshake
  ACTIONS = ['wink', 'double blink', 'close your eyes', 'jump']

  def initialize(number)
    @number = number
  end

  def commands
    return [] if @number.to_s =~ /[^\d]/

    actions = @number
      .to_s(2)
      .rjust(5, '0')
      .reverse
      .chars
      .each_with_index
      .inject ([]) {|acc, (v, index)|
        next acc if v == '0'
        acc << ACTIONS[index]
      }
    return actions.compact.reverse if actions.include? nil
    actions
  end
end

File: solutions/python/secret-handshake/1/secret_handshake.py
def commands(binary_str: str):
    actions = ['wink', 'double blink', 'close your eyes', 'jump']
    handshake = []
    for index, char in enumerate(binary_str[::-1]):
        if char == '1':
            if index > len(actions) - 1:
                handshake = handshake[::-1]
            else:
                handshake.append(actions[index])
    return handshake
